### PR TITLE
[NPUW]Ensure KV$ is left-aligned for all the chunks

### PR DESCRIPTION
**Pull request by DingZhangIntel**
### Details:
Align kv for all the chunks.

### Tickets:
[CVS-190643](https://jira.devtools.intel.com/browse/CVS-190643)

### AI Assistance:
 - *AI assistance used: no / yes*
 - *If yes, summarize how AI was used and what human validation was performed (build/tests/manual checks).*

### CODE REVIEW

**Comment by AlexanderKalistratov** [comment]
Left-aligned KV-cache is more complex topic than that.
It touches:
1. `PatchSlidingWindowMask` should be removed/disabled - trivial
2. Pyramid & HFA must be updated to support left-aligned KV-cache - simple
3. Detection of last token to pass to lm head - currently done on host, but it is possible only when lm head is cut out. Otherwise need to do dynamic detection in model itself - moderate
5. continuous prefill and probably prefix caching - rather trivial, but work is still in progress.
6. Fix alignment of number of inputs (like `per_layer_inputs`), not only kv-cache and `input_ids`.

But the main idea is that left-aligned KV-cache is actually touching a lot of places and a lot of things could go wrong. We need to consider if we want to introduce it by parts under the option. This means we would duplicate some part of codes which is not very well organized even now and introducing new option would simply doubles the complexity. 
Making all the changes in a single PR on the other hand is risky and hard to review.

File: src/plugins/intel_npu/src/plugin/npuw/attn/attn_subgraph.cpp [hunk 1]
```diff
@@ -858,13 +858,9 @@ ov::npuw::v1::subgraphs::RuntimeBehaviorFactory make_runtime_factory() {
...
-                            } else {
-                                const auto present_len = pyramid->get_context_length(pyramid_id) - past_len;
-                                copy_mask_segment(past_len, full_mask_shape[ATTN_KV_DIM] - present_len, present_len);
-                                copy_mask_segment(0, 0, past_len);
-                            }
+                            // Pyramid prefill is enabled only with chunked prefill. The staging mask
+                            // stores past and current tokens as one left-aligned context interval.
+                            copy_mask_segment(0, 0, pyramid->get_context_length(pyramid_id));
                             state.cached_attention_mask = dst;
                             return;
```
[the remark this answers is not in the document]
Comment (DingZhangIntel) [reply src/plugins/intel_npu/src/plugin/npuw/attn/attn_subgraph.cpp:869]: Fixed.

**Comment by DingZhangIntel** [comment]
The implementation has been split into three draft PRs for incremental review:

1. #37882: foundational left-aligned chunked-prefill KV cache and input staging
2. #37883: Pyramid/HFA attention consumers and SWA gating
3. #37884: shared and uncut LM-head final-token selection

The new branches are cumulative and the intended merge order is #37882 -> #37883 -> #37884. Closing this superseded combined PR to avoid duplicate review; all task changes remain in the three draft PRs above.

**Comment by intelgaoxiong** [comment]
Hi @DingZhangIntel 
Regarding to left kv$, I'm a little surprised that we don't need to change the mask generation?
 
Originally:
tokens: [padding] [past] [present]
mask:   [masked ] [   attended     ]
 
 
left kv$:
tokens:                [past][present] [padding ]
expected mask:   [   attended   ] [ masked ]
 
An intuitive feeling is mask should be different

**Comment by DingZhangIntel** [comment]
> Hi @DingZhangIntel Regarding to left kv$, I'm a little surprised that we don't need to change the mask generation?
> 
> Originally: tokens: [padding] [past] [present] mask: [masked ] [ attended ]
> 
> left kv$: tokens: [past][present] [padding ] expected mask: [ attended ] [ masked ]
> 
> An intuitive feeling is mask should be different

Thanks, your intuition is correct: the mask must follow the physical KV-cache layout.

Further changes have been split into the following stacked PRs:

1. #37882: foundational KV/input staging
2. #37883: Pyramid/HFA/SWA and auxiliary mask/input consumers
3. #37884: final-token selection for shared and uncut LM-head paths
